core/frontend/src/extension/Extension.ts

```diff
@@ -0,0 +1,56 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module Extensions
+ */
+
+/**
+ * @beta
+ */
+export type ResolveFunc = () => Promise<any>;
+
+/**
+ * @beta
+ */
+export type ActivationEvent = "onStartup";
```

Comment:
@kckst8 Do we need to expose this type?  It looks like we have an array for them at https://github.com/iTwin/itwinjs-core/pull/2743/files#diff-3ed6cce6e91b23ccf13b7553aad1615c9c0f3f6135fc8e0a59c3e5df116294c2R36.

Reply:
yeah we should decide on one for sure. I would prefer enums for this case. Thoughts?

Reply:
Agreed, I'd remove this type and export the Enum.  I'd mark it as `internal` as well since currently I don't see any reason an application would ever need it.